src/deep_compare.js

```diff
@@ -0,0 +1,22 @@
+export const deep_compare = (objectOne, objectTwo) => {
+  if (typeof objectOne !== 'object' || typeof objectTwo !== 'object') {
```

Comment:
if you do typeof null it will be 'object' as well.  I would also check whether objectOne and objectTwo are not null
```suggestion
  if (objectOne === null || typeof objectOne !== 'object' || objectTwo === null || typeof objectTwo !== 'object') {
```